provisioning/service/src/Query.cs

```diff
@@ -97,10 +97,7 @@ public class Query : IDisposable
                 throw new ArgumentException($"{nameof(pageSize)} cannot be negative.");
             }
 
-            // TODO: Refactor ContractApiHttp being created again
-            _contractApiHttp = new ContractApiHttp(
-                serviceConnectionString.HttpsEndpoint,
-                serviceConnectionString, httpTransportSettings);
```

Comment:
This block of code is the important bit here. Now we will pass around the single ```contractApiHttp``` instance everywhere instead of constructing a new one for each query

Reply:
Does this introduce any change with respect to processing limitations, i.e. will parallel requests on separate threads still execute asynchronously?

Reply:
No changes, no. We were already using the same httpClient to make every other request (createEnrollment, GetEnrollment, etc.) so this just brings the query requests in with them.